### PR TITLE
Docket branch new

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
Finally, it's done ;)

### CODE REVIEW

**Comment by Kathrin-Huber** [comment]
Fixes issue #627

File: Kitodo-Docket/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    </parent>
+    <name>Kitodo - Docket</name>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>kitodo-docket</artifactId>
+
+    <properties>
+        <maven.compiler.source>1.7</maven.compiler.source>
+        <maven.compiler.target>1.7</maven.compiler.target>
+    </properties>
```
Comment (henning-gerhardt) [inline Kitodo-Docket/pom.xml]: Compiler properties are not needed as they are defined in the global pom file and this values are wrong as we use Java 8 and not Java 7 for compiling.

File: Kitodo-Docket/src/test/java/org/kitodo/docket/DocketDataGenerator.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+package org.kitodo.docket;
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/java/org/kitodo/docket/DocketDataGenerator.java]: Missing license header.

File: Kitodo-Docket/src/test/java/org/kitodo/docket/DocketTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
+package org.kitodo.docket;
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/java/org/kitodo/docket/DocketTest.java]: Missing license header.

File: Kitodo-Docket/src/test/java/org/kitodo/docket/ExportDocketTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package org.kitodo.docket;
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/java/org/kitodo/docket/ExportDocketTest.java]: Missing license header.

File: Kitodo/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Arrays;
+import java.util.ServiceLoader;
+
+import org.kitodo.config.ConfigMain;
+
+public class KitodoServiceLoader<T> {
+    private ServiceLoader<T> loader;
+    private Class clazz;
+
+    public KitodoServiceLoader(Class clazz) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java]: All public methods without JavaDoc in this class.

File: Kitodo/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+                if (!urls.contains(udir)) {
+                    Class<URLClassLoader> sysClass = URLClassLoader.class;
+                    Method method = sysClass.getDeclaredMethod("addURL", URL.class);
+                    method.setAccessible(true);
+                    method.invoke(sysLoader, udir);
+                }
+            }
+        } catch (IOException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            e.printStackTrace();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java]: Do not use `printStackTrace()` - use logger mechanism.

File: Kitodo-Docket/src/test/resources/docket.xsl [hunk 7]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/resources/docket.xsl]: Please replace the old license header with new one!

File: Kitodo-Docket/src/test/resources/docket.xsl [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+ * distribute the resulting executable under terms of your choice, provided that you also meet, for each linked independent module, the terms and
+ * conditions of the license of that module. An independent module is a module which is not derived from or based on this library. If you modify this
+ * library, you may extend this exception to your version of the library, but you are not obliged to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ -->
+
+<xsl:stylesheet version="1.1" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:fo="http://www.w3.org/1999/XSL/Format"
+	xmlns:goobi="http://www.kitodo.org/logfile" exclude-result-prefixes="fo">
+	<xsl:output method="xml" indent="yes" />
+	<xsl:template match="goobi:process">
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/resources/docket.xsl]: Does this work as there was a renaming from "goobi" to "kitodo".

File: Kitodo-Docket/src/test/resources/docket.xsl [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+ * of the GNU General Public License cover the whole combination. As a special exception, the copyright holders of this library give you permission to
+ * link this library with independent modules to produce an executable, regardless of the license terms of these independent modules, and to copy and
+ * distribute the resulting executable under terms of your choice, provided that you also meet, for each linked independent module, the terms and
+ * conditions of the license of that module. An independent module is a module which is not derived from or based on this library. If you modify this
+ * library, you may extend this exception to your version of the library, but you are not obliged to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ -->
+
+<xsl:stylesheet version="1.1" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:fo="http://www.w3.org/1999/XSL/Format"
+	xmlns:goobi="http://www.kitodo.org/logfile" exclude-result-prefixes="fo">
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/resources/docket.xsl]: Tab indent must be changed to space indent in whole file

File: Kitodo-Docket/src/test/resources/docket_multipage.xsl [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ *
+-->
+
+<xsl:stylesheet version="1.1" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:fo="http://www.w3.org/1999/XSL/Format"
+	xmlns:kitodo="http://www.kitodo.org/logfile" exclude-result-prefixes="fo">
```
Comment (henning-gerhardt) [inline Kitodo-Docket/src/test/resources/docket_multipage.xsl]: Tab indent must be changed to space indent in the whole file.

**Comment by henning-gerhardt** [comment]
Function test:
- after running `mvn clean package` or `mvn clean install -Pall-tests` a new file `Kitodo-Docket/docket.pdf` is created. I think this file is created in tests but not removed afterwards.
- Creating a single docket in the application results in a download file called `batch_docket.pdf` instead of `<process_title>.pdf` as download file name. The name `batch_docket.pdf` for creating dockets from a batch is correct.

File: Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java [hunk 11]
```diff
@@ -1233,14 +1232,13 @@ public void downloadDocket(List<Process> processes) throws IOException {
...
                     xsltFile);
 
-            writeToOutputStream(facesContext, file);
-            facesContext.responseComplete();
+            writeToOutputStream(facesContext, file, "batch_docket.pdf");
         }
     }
 
-    private void writeToOutputStream(FacesContext facesContext, File file) throws IOException {
+    private void writeToOutputStream(FacesContext facesContext, File file, String filename) throws IOException {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/services/data/ProcessService.java]: if you already change it, you could use camelCases and this overwrite of value is not necessary

**Pull request merged**